### PR TITLE
[convectionkernels] Add port.

### LINKED ISSUE #25147: [New Port Request] cvtt
Library name: cvtt

Library description: Fast, high-quality texture compression library for many formats 

Source repository URL: https://github.com/elasota/ConvectionKernels

### PR DESCRIPTION
**Pull request by DragonJoker**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #25147

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Hopefully yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

### CODE REVIEW

File: ports/cvtt/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    EXPORT cvtt
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+install( EXPORT cvtt
+    FILE cvttConfig.cmake
+    COMPONENT cvtt
+    NAMESPACE cvtt::
+    DESTINATION lib/cmake/cvtt
```
Comment (Cheney-W) [inline ports/cvtt/CMakeLists.txt:81]: If upstream doesn't provide the targets, please export `unofficial-cvtt` and the namespace is `unofficial::cvtt::`

You could refer to the modifications in this [PR](https://github.com/microsoft/vcpkg/pull/25037/files).

File: ports/cvtt/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    ${${PROJECT_NAME}_HDR_FILES}
+    ${${PROJECT_NAME}_SRC_FILES}
+)
+add_library( cvtt::${PROJECT_NAME}
+    ALIAS
+    ${PROJECT_NAME}
+)
+target_include_directories( ${PROJECT_NAME}
+    PRIVATE
+        ${CMAKE_CURRENT_LIST_DIR}
```
Comment (Cheney-W) [inline ports/cvtt/CMakeLists.txt:54]:
```suggestion
        ${CMAKE_CURRENT_LIST_DIR}
        INTERFACE $<INSTALL_INTERFACE:include>
```

File: ports/cvtt/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "cvtt",
+  "version-date": "2022-06-08",
+  "description": "Fast, high-quality texture compression library for many formats ",
```
Comment (ras0219-msft) [inline ports/cvtt/vcpkg.json:4]:
```suggestion
  "name": "convectionkernels",
  "version-date": "2022-06-08",
  "description": "Fast, high-quality texture compression library for many formats, also known as cvtt",
```
We strongly prefer names longer than 5 characters (unless there's significant prior art). I also get much better SEO in search engines for "convectionkernels" (top hit is this library) versus (cvtt).

Adding it to the description makes sure that `vcpkg search cvtt` will still find it.

File: ports/cvtt/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_FakeBT709_Rounding.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_IndexSelector.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_IndexSelectorHDR.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_PackedCovarianceMatrix.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_ParallelMath.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_S3TC.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_S3TC_SingleColor.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_UnfinishedEndpoints.h
+    ${CMAKE_CURRENT_LIST_DIR}/ConvectionKernels_Util.h
+)
```
Comment (ras0219-msft) [inline ports/cvtt/CMakeLists.txt:39]:
```suggestion
file(GLOB SRC_FILES *.cpp)
file(GLOB HDR_FILES *.h)
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Two small changes, then LGTM.

Thanks for the PR!

**Pull request merged**